extensions/wikia/DMCARequest/DMCARequestSpecialController.class.php

```diff
@@ -0,0 +1,345 @@
+				$this->errorParam = $param;
+				return false;
+			}
+
+			$requestData[$param] = $paramValue;
+		}
+
+		$requestData['type'] = $request->getInt( 'type' );
+		if ( !$this->helper->isValidRequestorType( $requestData['type'] ) ) {
+			$this->error = $this->msg( 'dmcarequest-request-error-invalid' )->escaped();
+			$this->errorParam = 'type';
+			return false;
+		}
+
+		if ( !Sanitizer::validateEmail( $requestData['email'] ) ) {
+			$this->error = $this->msg( 'dmcarequest-request-error-invalid-email' )->escaped();
+			$this->errorParam = 'email';
+			return false;
+		}
+
+		$requestData['comments'] = trim( $request->getVal( 'comments' ) );
+
+		if ( !$request->getBool( 'goodfaith' )
+			|| !$request->getBool( 'perjury' )
+			|| !$request->getBool( 'wikiarights' )
+		) {
+			$this->error = $this->msg( 'dmcarequest-request-error-agreements' )->escaped();
+			return false;
+		}
+
+		$requestData['screenshots'] = $this->getScreenshots();
+
+		return $requestData;
+	}
+
+	/**
+	 * Get screenshots uploaded to the form.
+	 *
+	 * @return array An array of the screenshots suitable for
+	 *               attaching to the email.
+	 */
+	private function getScreenshots() {
+		$screenshots = $this->getContext()->getRequest()->getFileTempname( 'screenshots' );
+		$magic = MimeMagic::singleton();
+
+		$result = [];
+		if ( !empty( $screenshots ) ) {
+			foreach ( $screenshots as $image ) {
+				if ( !empty( $image ) ) {
+					$extList = '';
+					$mime = $magic->guessMimeType( $image );
+					if ( $mime !== 'unknown/unknown' ) {
+							// Get a space separated list of extensions
+							$extList = $magic->getExtensionsForType( $mime );
+							$fileExtension = strtok( $extList, ' ' );
+					} else {
+							$mime = 'application/octet-stream';
+					}
+					$result[] = [ 'file' => $image, 'ext' => $fileExtension, 'mime' => $mime ];
+				}
+			}
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Get the form HTML from the WikiaStyleGuideForm builder.
+	 *
+	 * @return string The form HTML
+	 */
+	private function getFormHtml() {
+		$request = $this->getRequest();
+		return $this->app->renderView( 'WikiaStyleGuideForm', 'index', [ 'form' => [
+			'attributes' => [ 'enctype' => 'multipart/form-data' ],
+			'isInvalid' => !empty( $this->error ),
+			'errorMsg' => $this->error,
+			'inputs' => [
+				[
+					'type' => 'select',
+					'name' => 'type',
+					'label' => $this->msg( 'dmcarequest-request-type-label' )->escaped(),
+					'options' => $this->getTypeOptions(),
+					'value' => Sanitizer::encodeAttribute( $request->getInt( 'type' ) ),
+				],
+				[
+					'type' => 'text',
+					'name' => 'fullname',
+					'isRequired' => true,
+					'attributes' => [ 'required' => 'required' ],
+					'label' => $this->msg( 'dmcarequest-request-name-label' )->escaped(),
+					'value' => Sanitizer::encodeAttribute( $request->getVal( 'fullname' ) ),
+					'isInvalid' => $this->errorParam === 'fullname',
+					'errorMsg' => $this->error,
+				],
+				[
+					'type' => 'textarea',
```

Comment:
Yep, because address inputs should be multiline to provide a properly formatted address (which matches the mockup of the form Brandon provided).